pkgs/by-name/README.md

```diff
@@ -1,9 +1,9 @@
 # Name-based package directories
 
 The structure of this directory maps almost directly to top-level package attributes.
-This is the recommended way to add new top-level packages to Nixpkgs [when possible](#limitations).
+This is how new top-level packages have to be added to Nixpkgs [when possible](#limitations).
```

Comment:
Add new top-level packages to Nixpkgs using this mechanism [whenever possible](#limitations).